sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -444,6 +528,28 @@ def process(self, batch, inference_args):
 
     return predictions
 
+  def process(
+      self, batch, inference_args, si_model_metadata: Optional[ModelMetdata]):
+    """
+    When side input is enabled:
+      The method checks if the side input model has been updated, and if so,
+      updates the model and runs inference on the batch of data. If the
+      side input is empty or the model has not been updated, the method
+      simply runs inference on the batch of data.
+    """
+    if si_model_metadata and self._enable_side_input_loading:
```

Comment:
Do we actually need to check `_enable_side_input_loading` here? Won't this always return the same value whether we have ` and self._enable_side_input_loading`?

Reply:
Or, alternatively can we ~just~ check `_enable_side_input_loading`?

Reply:
Yes, I want to use if `self._enable_side_input_loading` but mypy type checker is throwing errors since it interprets `si_model_metadata` as `Optional[ModelMetadata]`. So If I include `si_model_metadata`, the mypy errors goes off. 

We can just to if si_model_metadata as well. Let me see.

Reply:
Changed it.